app/controllers/application_controller.rb

```diff
@@ -1410,7 +1410,12 @@ def get_view(db, options = {}, fetch_data = false)
     @gtl_type = get_view_calculate_gtl_type(db_sym) unless fetch_data
 
     # Get the view for this db or use the existing one in the session
-    view = refresh_view ? get_db_view(db.gsub('::', '_'), :association => association, :view_suffix => view_suffix) : session[:view]
+    view =
+      if options['path_to_report']
+        MiqReport.new(YAML.load(File.open(options['path_to_report'])))
```

Comment:
This is suspicious .. 

either we're not sending the path on subsequent requests (which means sorting and paging will break),
or we are, but then we're sending the whole path, and opening whatever the user provided, without any checks that this path is still under ui-classic.

Reply:
Changed to sending only report name, not whole path.